builtin/providers/circonus/resource_circonus_check_http.go

```diff
@@ -0,0 +1,381 @@
+			},
+			checkHTTPVersionAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      defaultCheckHTTPVersion,
+				ValidateFunc: validateStringIn(checkHTTPVersionAttr, supportedHTTPVersions),
+			},
+		}, checkHTTPDescriptions),
+	},
+}
+
+// checkAPIToStateHTTP reads the Config data out of circonusCheck.CheckBundle into the
+// statefile.
+func checkAPIToStateHTTP(c *circonusCheck, d *schema.ResourceData) error {
+	httpConfig := make(map[string]interface{}, len(c.Config))
+
+	// swamp is a sanity check: it must be empty by the time this method returns
+	swamp := make(map[config.Key]string, len(c.Config))
+	for k, v := range c.Config {
+		swamp[k] = v
+	}
+
+	saveStringConfigToState := func(apiKey config.Key, attrName schemaAttr) {
+		if v, ok := c.Config[apiKey]; ok {
+			httpConfig[string(attrName)] = v
+		}
+
+		delete(swamp, apiKey)
+	}
+
+	saveIntConfigToState := func(apiKey config.Key, attrName schemaAttr) {
+		if v, ok := c.Config[apiKey]; ok {
+			i, err := strconv.ParseInt(v, 10, 64)
+			if err != nil {
+				panic(fmt.Sprintf("Unable to convert %s to an integer: %v", apiKey, err))
+			}
+
+			httpConfig[string(attrName)] = int(i)
+		}
+
+		delete(swamp, apiKey)
+	}
+
+	saveStringConfigToState(config.AuthMethod, checkHTTPAuthMethodAttr)
+	saveStringConfigToState(config.AuthPassword, checkHTTPAuthPasswordAttr)
+	saveStringConfigToState(config.AuthUser, checkHTTPAuthUserAttr)
+	saveStringConfigToState(config.Body, checkHTTPBodyRegexpAttr)
+	saveStringConfigToState(config.CAChain, checkHTTPCAChainAttr)
+	saveStringConfigToState(config.CertFile, checkHTTPCertFileAttr)
+	saveStringConfigToState(config.Ciphers, checkHTTPCiphersAttr)
+	saveStringConfigToState(config.Code, checkHTTPCodeRegexpAttr)
+	saveStringConfigToState(config.Extract, checkHTTPExtractAttr)
+
+	headers := make(map[string]interface{}, len(c.Config))
+	headerPrefixLen := len(config.HeaderPrefix)
+	for k, v := range c.Config {
+		if len(k) <= headerPrefixLen {
+			continue
+		}
+
+		if strings.Compare(string(k[:headerPrefixLen]), string(config.HeaderPrefix)) == 0 {
+			key := k[headerPrefixLen:]
+			headers[string(key)] = v
+		}
+		delete(swamp, k)
+	}
+	httpConfig[string(checkHTTPHeadersAttr)] = headers
+
+	saveStringConfigToState(config.KeyFile, checkHTTPKeyFileAttr)
+	saveStringConfigToState(config.Method, checkHTTPMethodAttr)
+	saveStringConfigToState(config.Payload, checkHTTPPayloadAttr)
+	saveIntConfigToState(config.ReadLimit, checkHTTPReadLimitAttr)
+	saveStringConfigToState(config.URL, checkHTTPURLAttr)
+	saveStringConfigToState(config.HTTPVersion, checkHTTPVersionAttr)
+
+	whitelistedConfigKeys := map[config.Key]struct{}{
+		config.ReverseSecretKey: struct{}{},
+		config.SubmissionURL:    struct{}{},
+	}
+
+	for k := range swamp {
+		if _, ok := whitelistedConfigKeys[k]; ok {
+			delete(c.Config, k)
+		}
+
+		if _, ok := whitelistedConfigKeys[k]; !ok {
+			panic(fmt.Sprintf("PROVIDER BUG: API Config not empty: %#v", swamp))
```

Comment:
I admit I don't recall all the specifics of our video chat the other week, but I don't believe `panic` is appropriate here. Do you recall if we covered that? 

In generally I default to logging unexpected things and continuing on (or perhaps erring). I feel a panic would be a very bad thing for operators to receive.